src/api/customRuns.js

```diff
@@ -39,6 +42,53 @@ function getCustomRunsAPI({ filters, isWebSocket, name, namespace }) {
   );
 }
 
+export function getCustomRunPayload({
+  kind,
+  labels,
+  namespace,
+  nodeSelector,
+  params,
+  serviceAccount,
+  customName,
+  customRunName = `${customName ? `${customName}-run` : 'run'}-${Date.now()}`,
+  timeout
+}) {
+  const payload = {
+    apiVersion: 'tekton.dev/v1beta1',
+    kind: 'CustomRun',
+    metadata: {
+      name: customRunName,
+      namespace
+    },
+    spec: {
+      customRef: {
+        apiVersion: '',
+        kind: 'Custom'
+      }
+    }
+  };
+  if (labels) {
+    payload.metadata.labels = labels;
+  }
+  if (params) {
+    payload.spec.params = Object.keys(params).map(name => ({
+      name,
+      value: params[name]
+    }));
+  }
+  if (nodeSelector) {
+    payload.spec.podTemplate = { nodeSelector };
+  }
```

Comment:
As mentioned in [a previous review](https://github.com/tektoncd/dashboard/pull/3108#discussion_r1332042350), podTemplate and nodeSelector are not valid for a CustomRun so should be removed:
```suggestion
```